python/hail/methods/io.py

```diff
@@ -0,0 +1,120 @@
+
+    Examples
+    --------
+    Export to VCF as a block-compressed file:
+
+    >>> methods.export_vcf(dataset, 'output/example.vcf.bgz')
+
+    Notes
+    -----
+    :meth:`export_vcf` writes the dataset to disk in VCF format as described in the
+    `VCF 4.2 spec <https://samtools.github.io/hts-specs/VCFv4.2.pdf>`__.
+
+    Use the ``.vcf.bgz`` extension rather than ``.vcf`` in the output file name
+    for `blocked GZIP <http://www.htslib.org/doc/tabix.html>`__ compression.
+
+    Note
+    ----
+    We strongly recommended compressed (``.bgz`` extension) and parallel
+    output (`parallel` set to ``'separate_header'`` or
+    ``'header_per_shard'``) when exporting large VCFs.
+
+    Hail exports the fields of Struct `info` as INFO fields,
+    the elements of Set[String] `filters` as FILTERS, and the
+    value of Float64 `qual` as QUAL. No other row fields are exported.
+
+    The FORMAT field is generated from the entry schema, which
+    must be a :py:class:`~hail.expr.TStruct`.  There is a FORMAT
+    field for each field of the Struct.
+
+    INFO and FORMAT fields may be generated from Struct fields of type Call,
+    Int32, Float32, Float64, or String. If a field has type Int64, every value
+    must be a valid Int32. Arrays and Sets containing these types are also
+    allowed but cannot be nested; for example, Array[Array[Int32]] is invalid.
+    Sets and Arrays are written with the same comma-separated format. Boolean
+    fields are also permitted in `info` and will generate INFO fields of
+    VCF type Flag.
+
+    Hail also exports the name, length, and assembly of each contig as a VCF
+    header line, where the assembly is set to the :class:`.GenomeReference`
+    name.
+
+    Consider the workflow of importing a VCF and immediately exporting the
+    dataset back to VCF. The output VCF header will contain FORMAT lines for
+    each entry field and INFO lines for all fields in `info`, but these lines
+    will have empty Description fields and the Number and Type fields will be
+    determined from their corresponding Hail types. To output a desired
+    Description, Number, and/or Type value in a FORMAT or INFO field or to
+    specify FILTER lines, use the `metadata` parameter to supply a dictionary
+    with the relevant information. See
+    :py:class:`~hail.api2.HailContext.get_vcf_metadata` for how to obtain the
+    dictionary corresponding to the original VCF, and for info on how this
+    dictionary should be structured. 
+    
+    The output VCF header will also contain CONTIG lines
+    with ID, length, and assembly fields derived from the reference genome of
+    the dataset.
+    
+    The output VCF header will `not` contain lines added by external tools
+    (such as bcftools and GATK) unless they are explicitly inserted using the
+    `append_to_header` parameter.
+
+    Warning
+    -------
+    Hail makes no assumptions about the meaning of `info` fields. In
```

Comment:
confusing wording: "do not automatically reflect modifications to the dataset".

Prefer something like:

> Hail exports the info field as-is. In particular, any values stored when a VCF was imported are not automatically modified to reflect filtered samples or genotypes, which can affect info fields like AC (allele count), AF (allele frequency), AN (allele number), etc. If a filtered dataset is export to VCF without preprocessing, downstream tools which...